src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,86 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
```

Comment:
Interesting. I'd forgotten that the code before the first yield doesn't occur until enumeration begins.

It looks like Unix and Win32 get around this with their custom enumerable class, but I'd really prefer to avoid that. How about if I move the yielding to a separate method e.g.

```
        public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
        {
            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
            return TranslatePaths(storageFiles);            
        }

        public static IEnumerable<string> TranslatePaths(IEnumerable<IStorageItem> storageFiles)
        {
            // Yield the paths of the StorageItems
            foreach (IStorageItem item in storageFiles)
            {
                yield return item.Path;
            }
        }
```

That way we do the file entry enumeration at initialization and the translation method is the one doing the yielding.